app/src/main/java/com/mybus/activity/DisplayFavoritesActivity.java

```diff
@@ -20,4 +103,90 @@ public int getToolbarId() {
     protected int getToolbarTittle() {
         return R.string.displayFavoritesToolbarTittle;
     }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        switch (resultCode) {
+            case RESULT_CANCELED:
+                //The user canceled
+                break;
+            case RESULT_OK:
+                newAddress = data.getStringExtra(SearchActivity.RESULT_STREET_EXTRA);
+                newLocation = data.getParcelableExtra(SearchActivity.RESULT_LATLNG_EXTRA);
+                FavoriteNameAlertDialog favoriteNameAlertDialog = new FavoriteNameAlertDialog();
+                switch (requestCode) {
+                    case ADD_SEARCH_RESULT_ID:
+                        favoriteNameAlertDialog.setActionAdding(this);
+                        favoriteNameAlertDialog.setDialogType(favoriteNameAlertDialog.TYPE_ADD);
```

Comment:
It's Defined as: `public static final String TYPE_ADD = "TYPE_ADD";` Why do you use the instance for this constant?